shell/pages/c/_cluster/fleet/settings/index.vue

```diff
@@ -0,0 +1,197 @@
+          weight:  2,
+          type:    'number',
+          path:    'helmops.replicas',
+          default: 1,
+        },
+
+        // 'agent' group
+        agentReplicas: {
+          weight:  0,
+          type:    'number',
+          path:    'agent.replicas',
+          default: 1,
+        },
+        agentTLSMode: {
+          weight: 1,
+          type:   'string',
+          items:  [{
+            type:  'string',
+            value: 'strict'
+          }, {
+            type:  'string',
+            value: 'system-store'
+          }],
+          path:    'agentTLSMode',
+          default: 'system-store',
+        },
+        agentCheckinInterval: {
+          weight:  2,
+          type:    'string',
+          path:    'agentCheckinInterval',
+          default: '15m',
+        },
+
+        // 'resource-definition' group
+        priorityClassName: {
+          weight:  0,
+          type:    'string',
+          path:    'priorityClassName',
+          default: '',
+        },
+        nodeSelector: {
+          weight:  1,
+          type:    'object',
+          path:    'nodeSelector',
+          default: {},
+        },
+        tolerations: {
+          weight:  2,
+          type:    'object',
+          path:    'tolerations',
+          default: [],
+        },
+        extraAnnotations: {
+          weight:  3,
+          type:    'object',
+          path:    'extraAnnotations',
+          default: {},
+        },
+
+        // 'developer' group
+        metrics: {
+          weight:  0,
+          type:    'boolean',
+          path:    'metrics.enabled',
+          default: false,
+        },
+        debug: {
+          weight:  1,
+          type:    'boolean',
+          path:    'debug',
+          default: false,
+        }
+      };
+    },
+
+    groups(): Group[] {
+      return [{
+        name:     'general',
+        children: [
+          'garbageCollectionInterval',
+          'gitClientTimeout',
+          'proxy',
+          'noProxy',
+          'clusterEnqueueDelay'
+        ],
+        weight: 0
+      }, {
+        name:     'controller',
+        children: [
+          'controllerReplicas',
+          'gitjobReplicas',
+          'helmopsReplicas',
+        ],
+        weight: 1
+      }, {
+        name:     'agent',
+        children: [
+          'agentReplicas',
+          'agentTLSMode',
+          'agentCheckinInterval'
+        ],
+        weight: 2
+      }, {
+        name:     'modifiers',
+        children: [
+          'nodeSelector',
+          'priorityClassName',
+          'tolerations',
+          'extraAnnotations'
+        ],
+        weight: 3
+      }, {
+        name:     'developer',
+        children: [
+          'metrics',
+          'debug'
+        ],
+        weight: 4
+      }];
+    }
+  }
+};
+</script>
+
+<template>
+  <ConfigMapSettings
+    :settings="settings"
+    :groups="groups"
+    :namespace="'cattle-system'"
+    :name="'rancher-config'"
```

Comment:
I would definitely rename the name from `rancher-config` which is quite generic... something that provides more context like `rancher-fleet-settings` or something